client/web/src/enterprise/batches/settings/github-apps-filter.ts

```diff
@@ -0,0 +1,22 @@
+import { BatchChangesCodeHostFields, ExternalServiceKind } from '../../../graphql-operations'
+
+// This function helps us figure out if an installation is in progress, until we have a more permanent state
+// via SRCH-798. Once that issues is closed, this function can maybe be dropped.
+export function credentialForGitHubAppExists(
+    appName: string | null,
+    nodes: BatchChangesCodeHostFields[] | undefined
+): boolean {
+    if (!appName) {
+        return false
+    }
+
+    if (!nodes || nodes.length === 0) {
+        return false
+    }
+
+    return (
+        nodes.filter(
+            n => n.externalServiceKind === ExternalServiceKind.GITHUB && n.credential?.gitHubApp?.name === appName
+        ).length > 0
+    )
```

Comment:
This works, but we could opt for a cool array iterator available in javascript.

```suggestion
    return !nodes.some(n => n.externalServiceKind === ExternalServiceKind.GITHUB && n.credential?.gitHubApp?.name === appName)
```

You can read about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).

Reply:
Right! `some` works too in that case.

Reply:
Yeah, just a suggestion. Feel free to discard / put off to a later PR.
THe performance improvement is negligible for now as we dont have customers with a bucket load of codehost connections.